src/storage/mvcc/txn.rs

```diff
@@ -266,7 +275,9 @@ impl<S: Snapshot> MvccTxn<S> {
                 }
             }
             _ => {
-                return match self.reader.get_txn_commit_info(&key, self.start_ts)? {
+                let (txn_commit_info, write_ts_collision) =
+                    self.reader.get_txn_commit_info(&key, self.start_ts)?;
```

Comment:
Can we use two function to do such complicated job? Since this branch is not hot path.